Wrappers/Python/cil/processors/Binner.py

```diff
@@ -14,218 +14,410 @@
 #   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
+
 from cil.framework import DataProcessor, AcquisitionData, ImageData, DataContainer, AcquisitionGeometry, ImageGeometry
-import numpy
-import warnings
+import numpy as np
+import weakref
+
+try:
+    from cil.processors.cilacc_binner import Binner_IPP
+    has_ipp = True
+except:
+    has_ipp = False
 
 
 class Binner(DataProcessor):
-    r'''Binner processor rebins (downsample) array and returns new geometry.
-        
-    :param roi: region-of-interest to bin, specified as a dictionary containing tuple (Start, Stop, Step)
-    :type roi: dict
-    :return: returns an AcquisitionData or ImageData object with an updated AcquisitionGeometry or ImageGeometry
-    :rtype: AcquisitionData or ImageData
-    '''
-    
-    '''
-    Start inclusive, Stop exclusive
-    
-    -1 is a shortcut to include all elements along the specified dimension
-    
-    if only one number is provided, then it is interpreted as Stop
-    
-    if two numbers are provided, then they are interpreted as Start and Stop
+
+    """This creates a Binner processor.
     
-    Start = None is equivalent to Start = 0
-    Stop = None is equivalent to Stop = number of elements
-    Step = None is equivalent to Step = 1
+    The processor will crop the data, and then average together n input pixels along a dimension from the starting index.
+
+    The output will be a data container with the data, and geometry updated to reflect the operation.
+
+    Parameters
+    ----------
+
+    roi : dict
+        The region-of-interest to bin {'axis_name1':(start,stop,step), 'axis_name2':(start,stop,step)}
+        The `key` being the axis name to apply the processor to, the `value` holding a tuple containing the ROI description
+
+        Start: Starting index of input data. Must be an integer, or `None` defaults to index 0.
+        Stop: Stopping index of input data. Must be an integer, or `None` defaults to index N.
+        Step: Number of pixels to average together. Must be an integer or `None` defaults to 1.
+
+    accelerated : boolean, default=True
+        Used the CIL accelerated backend if `True`, numpy if `False`.
+
+
+    Example
+    -------
+    from cil.processors import Binner
+    roi = {'horizontal':(10,-10,2),'vertical':(10,-10,2)}
+    processor = Binner(roi)
+    processor.set_input(data)
+    data_binned = processor.get_output()
+
+
+    Example
+    -------
+    from cil.processors import Binner
+    roi = {'horizontal':(10,-10,2),'vertical':(10,-10,2)}
+    processor = Binner(roi)
+    processor.set_input(data.geometry)
+    geometry_binned = processor.get_output()
```

Comment:
The examples are not easy to read on the documentation website:
![image](https://user-images.githubusercontent.com/60604372/215073134-9c07f6bf-1659-4ef7-a292-80758e060cb1.png)

In the example docstring here: https://numpydoc.readthedocs.io/en/latest/example.html#example you can scroll down to see that the code is rendered nicely. They mention the doctest format https://www.digitalocean.com/community/tutorials/how-to-write-doctests-in-python which means you need to use >>> in front of any code lines. I don't know if you also need an empty line before any code